ui/lib/kv/addon/components/page/secret/metadata/version-diff.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { kvDataPath } from 'vault/utils/kv-path';
+import { typeOf } from '@ember/utils';
+
+/**
+ * @module KvVersionDiff
+ * This component produces a JSON diff view between 2 secret versions. It uses the library jsondiffpatch.
+ *
+ * @param {string} backend - Backend from the kv/data model.
+ * @param {string} path - Backend from the kv/data model.
+ * @param {array} metadata - The kv/metadata model. It is version agnostic.
+ * @param {array} breadcrumbs - Array to generate breadcrumbs, passed to the page header component.
+ */
+
+export default class KvVersionDiffComponent extends Component {
+  @service store;
+
+  @tracked leftSideVersion = this.args.metadata.currentVersion;
+  @tracked rightSideVersion = this.defaultRightSideVersion;
```

Comment:
I think tracked properties typically should be initially set in the constructor?